src/process_utility.c

```diff
@@ -975,6 +977,12 @@ process_vacuum(ProcessUtilityArgs *args)
 															  cp->compressed_relid);
 		}
 	}
+	/*
+	Restore original list. stmt->rels which has references to
+	VacuumRelation list is freed up, however VacuumStmt is not
+	cleaned up because of which there is a crash.
+	*/
+	stmt->rels = list_copy(stmt_rels);
```

Comment:
I am curious, do you know in which places `stmt->rels` is changed? Is there any change besides the one in line 963 (`stmt->rels = list_concat(ctx.chunk_rels, vacuum_rels);`). 

Maybe we could also fix the problem by creating a new list there and using it for the iteration a few lines later. Then we would save the copy and restore operation and `stmt->rels` would always contain valid entries.

Reply:
You are right. We can fix it either way.
I tried the same approach initially. I created a new list instead of modifying stmt->rels, but i could not pass this new list to ExecVacuum(), as this API expects VacuumStmt as 2nd parameter. Which in turn requires me to create a new VacuumStmt object and set the new list inside it.

Thus i took alternative approach of saving the original list, and restoring the saved list.

Reply:
Isn't this similar to this? https://github.com/timescale/timescaledb/commit/74ca546565076676b4b18af3d369465dea4147e3